x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldType.java

```diff
@@ -174,46 +180,115 @@ public SpanQuery spanPrefixQuery(String value, SpanMultiTermQueryWrapper.SpanRew
     private String unsupported(String query, String supported) {
         return String.format(
             Locale.ROOT,
-            "Can only use %s queries on %s fields - not on [%s] which is of type [%s] with runtime_type [%s]",
+            "Can only use %s queries on %s fields - not on [%s] which is a runtime field of type [%s]",
             query,
             supported,
             name(),
-            RuntimeFieldMapper.CONTENT_TYPE,
-            runtimeType()
+            typeName()
         );
     }
 
     protected final void checkAllowExpensiveQueries(QueryShardContext context) {
         if (context.allowExpensiveQueries() == false) {
             throw new ElasticsearchException(
-                "queries cannot be executed against ["
-                    + RuntimeFieldMapper.CONTENT_TYPE
-                    + "] fields while ["
-                    + ALLOW_EXPENSIVE_QUERIES.getKey()
-                    + "] is set to [false]."
+                "queries cannot be executed against runtime fields while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
             );
         }
     }
 
-    /**
-     * The format that this field should use. The default implementation is
-     * {@code null} because most fields don't support formats.
-     */
-    protected String format() {
-        return null;
+    @Override
+    public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup lookup, String format) {
+        return new DocValueFetcher(docValueFormat(format, null), lookup.doc().getForField(this));
+    }
+
+    @Override
+    protected final void doXContentBody(XContentBuilder builder, boolean includeDefaults) throws IOException {
+        toXContent.accept(builder, includeDefaults);
     }
 
     /**
-     * The locale that this field's format should use. The default
-     * implementation is {@code null} because most fields don't
-     * support formats.
+     *  For runtime fields the {@link RuntimeFieldType.Parser} returns directly the {@link org.elasticsearch.index.mapper.MappedFieldType}.
+     *  Internally we still create a {@link Builder} so we reuse the {@link FieldMapper.Parameter} infrastructure,
+     *  but {@link Builder#init(FieldMapper)} and {@link Builder#build(ContentPath)} are never called as
+     *  {@link RuntimeFieldTypeParser#parse(String, Map, Mapper.TypeParser.ParserContext)} calls
+     *  {@link Builder#parse(String, Mapper.TypeParser.ParserContext, Map)} and returns the corresponding
+     *  {@link org.elasticsearch.index.mapper.MappedFieldType}.
      */
```

Comment:
ehm, doing what?

Reply:
Using the Builder at all. It feels like it makes sense because we have them now, but maybe its overkill?

Reply:
I like using Parameters to define what gets parsed and to reuse that part of the existing mappings infra. I think that not using them would mean going back to manual parsing and printing.

Reply:
Me too! I just wonder if we can avoid building the FieldMapper then. Like, in a follow up? Or not. I dunno. It just fields weird to build a dummy field mapper to make parsing easier. No big deal.

Reply:
we do not build the field mapper at all!

Reply:
pressed enter too quickly: we skip calling build on the builder. we make the builder create the right field type and return that, without creating the field mapper.

Reply:
one follow-up we could thin of to clean things up is to extract the parsing bits from FieldMapper.Builder as well as the Parameter class so that parsing can be done outside of a Builder.